mlir-tensorrt/compiler/test/Pipelines/TensorRTClustering/tensorrt-clustering.mlir

```diff
@@ -1,15 +1,28 @@
 // RUN: mlir-tensorrt-opt %s -tensorrt-clustering-pipeline -split-input-file | FileCheck %s
 
-func.func @trt_relu(%arg0: tensor<2x10xf16>) -> (tensor<2x10xf16>) {
+func.func @trt_relu(%arg0: tensor<?xf16> {tensorrt.shape_profile = #tensorrt.shape_profile<min=[1],opt =[4],max=[7]>}) -> (tensor<?xf16>) {
   %0 = tensorrt.activation {
     activationType = #tensorrt.activation_type<kRELU>
-  } %arg0 : tensor<2x10xf16>
-  return %0: tensor<2x10xf16>
+  } %arg0 : tensor<?xf16>
+  return %0: tensor<?xf16>
 }
 
 // CHECK-LABEL: @trt_relu
 // CHECK-DAG:   %[[v0:.+]] = tensorrt.call_alloc @trt_engines::@tensorrt_cluster
 // CHECK-DAG: return %[[v0]]
-// CHECK-DAG: @tensorrt_cluster
+// CHECK-DAG: @tensorrt_cluster(%arg0: tensor<?xf16> {tensorrt.shape_profile = #tensorrt.shape_profile<min = [1], opt = [4], max = [7]>})
 // CHECK-DAG: %[[v1:.+]] = tensorrt.activation {activationType = #tensorrt.activation_type<kRELU>}
 // CHECK-DAG: return %[[v1]]
+
+// -----
```

Comment:
To head off any future problems, you may want to add a more complicated test case here where the function has multiple arguments. Some variations might be: N arguments where a couple are unused, have the arguments be used in an order other than their list order (to try to create a clustering where the args to the cluster have a different order). 

You can also create a negative test case. Add `-verify-diagnostics` to the RUN command, then you can have a test like


```
// expected-error @below {{expected error message}}
func.func @no_shape_profile(%arg0: tensor<?xf32>) -> tensor<?xf32> {
   // body just like `trt_relu` test
   ...
 }
```